### PR TITLE
Support history file in REPL

### PR DESCRIPTION
**Pull request by shendiaomo**

### CODE REVIEW

**Comment by tonyyang-svail** [comment]
Hi @shendiaomo, could you please add tests for this package?

**Comment by shendiaomo** [comment]
> Hi @shendiaomo, could you please add tests for this package?

Sure. There is already a `repl_test.go` in the package that covers most of the complex functions. I'll add some test cases later.

File: cmd/repl/prompt.go [hunk 1]
```diff
@@ -167,6 +173,32 @@ func (p *promptState) initCompleter() {
...
+	defer f.Close()
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		p.history = append(p.history, scanner.Text())
+	}
+}
+
+func (p *promptState) updateHistory() {
+	if p.statement != "" {
+		f, err := os.OpenFile(p.historyFileName, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0644)
```
Comment (shendiaomo) [inline cmd/repl/prompt.go:191]: Posix seems to guarantee that writes with O_APPEND are atomic. The APUE book also describes the behavior in detail.  So we don't need a file lock here.

**Review by weiguoz — APPROVED** [review]
LGTM

**Pull request merged**